ddtrace/internal/_encoding.pyx

```diff
@@ -0,0 +1,357 @@
+            elif PyInt_CheckExact(o):
+                longval = o
+                ret = msgpack_pack_long(&self.pk, longval)
+            elif PyFloat_CheckExact(o):
+                if self.use_float:
+                   fval = o
+                   ret = msgpack_pack_float(&self.pk, fval)
+                else:
+                   dval = o
+                   ret = msgpack_pack_double(&self.pk, dval)
+            elif PyBytesLike_CheckExact(o):
+                L = len(o)
+                if L > ITEM_LIMIT:
+                    PyErr_Format(ValueError, b"%.200s object is too large", Py_TYPE(o).tp_name)
+                rawval = o
+                ret = msgpack_pack_bin(&self.pk, L)
+                if ret == 0:
+                    ret = msgpack_pack_raw_body(&self.pk, rawval, L)
+            elif PyUnicode_CheckExact(o):  #  if strict_types else PyUnicode_Check(o):
+                if self.encoding == NULL:
+                    ret = msgpack_pack_unicode(&self.pk, o, ITEM_LIMIT)
+                    if ret == -2:
+                        raise ValueError("unicode string is too large")
+                else:
+                    o = PyUnicode_AsEncodedString(o, self.encoding, self.unicode_errors)
+                    L = len(o)
+                    if L > ITEM_LIMIT:
+                        raise ValueError("unicode string is too large")
+                    ret = msgpack_pack_raw(&self.pk, L)
+                    if ret == 0:
+                        rawval = o
+                        ret = msgpack_pack_raw_body(&self.pk, rawval, L)
+            elif PyDict_CheckExact(o):
+                d = <dict>o
+                L = len(d)
+                if L > ITEM_LIMIT:
+                    raise ValueError("dict is too large")
+                ret = msgpack_pack_map(&self.pk, L)
+                if ret == 0:
+                    for k, v in d.items():
+                       ret = self._pack(k)
+                       if ret != 0: break
+                       ret = self._pack(v)
+                       if ret != 0: break
+            elif PyList_CheckExact(o):
+                # Expect a list of traces or a list of spans
+
+                L = len(o)
+                if L > ITEM_LIMIT:
+                    raise ValueError("list is too large")
+
+                ret = msgpack_pack_array(&self.pk, L)
+                if ret != 0:
+                    break
+
+                if L > 0 and PyList_CheckExact(o[0]):
+                    # List of traces
+                    for i in range(L):
+                        span = o[i]
+                        ret = self._pack(span)
+                        if ret != 0: break
+                else:
+                    # List of spans
+                    for i in range(L):
+                        ret = self._pack(o[i])
+                        if ret != 0: break
+
+            elif type(o) is Span:
+                L = 12
+                if L > ITEM_LIMIT:
```

Comment:
will this ever fail?

Reply:
does this need to be at L218?

Reply:
lol almost certainly won't fail... probably silly to keep it around. It would make more sense to be at 218 but I think we can just do away with it